Doc/reference/datamodel.rst

```diff
@@ -2363,11 +2363,13 @@ left undefined.
    These methods are called to implement the binary arithmetic operations
    (``+``, ``-``, ``*``, ``@``, ``/``, ``//``, ``%``, :func:`divmod`,
    :func:`pow`, ``**``, ``<<``, ``>>``, ``&``, ``^``, ``|``) with reflected
-   (swapped) operands.  These functions are only called if the left operand does
-   not support the corresponding operation [#]_ and the operands are of different
-   types. [#]_ For instance, to evaluate the expression ``x - y``, where *y* is
+   (swapped) operands.  These functions are only called if the operands
+   are of different types, when the left operand does not support the corresponding
+   operation [#]_ or the right operand's class is derived from the left operand's
+   class. [#]_ For instance, to evaluate the expression ``x - y``, where *y* is
    an instance of a class that has an :meth:`__rsub__` method, ``y.__rsub__(x)``
-   is called if ``x.__sub__(y)`` returns *NotImplemented*.
+   is called if ``x.__sub__(y)`` returns *NotImplemented* or ``type(y)`` is a
+   subclass of ``type(x)``.
 
```

Comment:
> These functions are only called if the operand types are different.  If the type of the right operand is a subclass of the left operand's type, then the right operand's reflected method will be called first [foot note here]; if that method returns *NotImplemented* then the left operand's normal method will be called.  If the right operand's type is not a subclass of the left's type,  then the left operand's normal method will be called first, and only if it returns *NotImplemented* will the right operand's reflected method be called.

> For instance, to evaluate the expression ``x - y``, where *y* is an instance of a class that has an :meth:`__rsub__` method, ``y.__rsub__(x)``  will be called first if ``type(y)`` is a subclass of ``type(x)``, and called second if ``type(y)`` is not a subclass of ``type(x)``.

Reply:
@ethanfurman  Hmm, with your suggestion, there are two existing footnotes which become orphaned. https://github.com/python/cpython/blame/a33f2c2bae759fc9d06e1c032fd2026135f2df45/Doc/reference/datamodel.rst#L2768-L2776  
So I'm not sure what to do here. Those notes look useful.

Reply:
Those notes are useful, thanks for catching that.  I edited the above to include them.